OpenRA.Mods.Common/UtilityCommands/UpgradeRules.cs

```diff
@@ -185,6 +185,23 @@ internal static void UpgradeActorRules(ModData modData, int engineVersion, ref L
 					}
 				}
 
+				// GivesExperience was changed from having a FriendlyFire bool to a Stances array.
+				if (engineVersion < 20160702 && node.Key.StartsWith("GivesExperience"))
+				{
+					var ff = "FriendlyFire";
+					var ffNode = node.Value.Nodes.FirstOrDefault(n => n.Key == ff);
+					if (ffNode != null)
+					{
+						if (FieldLoader.GetValue<bool>(ff, ffNode.Value.Value))
```

Comment:
Then the node is removed and that is the end of the upgrade rule, leaving them with `Neutral` and `Enemy`.